### PR TITLE
Ensdf processing logft

### PR DESCRIPTION
**Pull request by howland**
Adds logft, radd, and ruler ensdf evaluation tools to PyNE.

### CODE REVIEW

**Comment by scopatz** [comment]
Looks like we have some CI failures.

**Comment by howland** [comment]
@scopatz I'm still working on getting one of the tests I wrote to pass.

**Comment by scopatz** [comment]
Ok! Just ping when it is ready then.

**Comment by howland** [comment]
@scopatz Its now ready for review, I ended up removing the evaluation tool that wasn't use-able (yet) in Python3.

File: pyne/ensdf_processing.py [hunk 1]
```diff
@@ -104,3 +104,105 @@ def gabs(inputdict_unchecked):
     proc.communicate()[0]
     proc.stdin.close()
 
+def logft(inputdict_unchecked):
+    #NOTE: changed input file line length to 90 to support longer file paths in fortran source.
+    """
+    This program calculates log ft values for beta and electron-capture decay, average beta energies, 
+    and capture fractions.  (LOGFT readme)
+
+    Input Dictionary Required Key Pair Value:
```
Comment (scopatz) [inline pyne/ensdf_processing.py]: Please use numpydoc format

File: pyne/ensdf_processing.py [hunk 2]
```diff
@@ -104,3 +104,105 @@ def gabs(inputdict_unchecked):
...
+    """
+    inputdict = {}
+    input_data_set = inputdict_unchecked['input_data_set']
+    output_report = inputdict_unchecked['output_report']
+    data_table = inputdict_unchecked['data_table']
+    output_data_set = inputdict_unchecked['output_data_set']
+
+    exe_path = path_to_exe('logft')
+    inp = input_data_set + '\n' + output_report + '\n' + data_table + '\n' + output_data_set + '\n'
+    proc = subprocess.Popen([exe_path],stdout=subprocess.PIPE,stdin=subprocess.PIPE)
```
Comment (scopatz) [inline pyne/ensdf_processing.py:145]: `subprocess.check_output()` with the `input` kwarg is a nicer options for this and the following lines.  Not a blocker, but unless you really need it you probably shouldn't be using Popen() directly.

File: pyne/ensdf_processing.py [hunk 3]
```diff
@@ -104,3 +104,105 @@ def gabs(inputdict_unchecked):
...
+    Input Dictionary Required Key Pair Value:
+        input_data_set : path to input data file.
+        output_report : desired path to output report file.
+        data_table : path to data table.
+        output_data_set : desired path to output data set.
+
+    Output Dictionary Values:
+        Everything in input dictionary is returned if LOGFT completes successfully.
+    """
+    inputdict = {}
```
Comment (scopatz) [inline pyne/ensdf_processing.py:137]: As a general point, though not necessarily a blocker, in Python the convention of including the type name in the variable name is not advised.  Not a huge deal, but something to be aware of.

File: pyne/ensdf_processing.py [hunk 4]
```diff
@@ -104,3 +104,105 @@ def gabs(inputdict_unchecked):
...
+
+    exe_path = path_to_exe('radd')
+    inp = atomic_number + '\n' + neutron_number + '\n' + 'NO' + '\n'
+    proc = subprocess.Popen([exe_path],stdout=subprocess.PIPE,stdin=subprocess.PIPE)
+    proc.stdin.write(inp.encode('utf-8'))
+    radd_output = proc.communicate()[0]
+    proc.stdin.close()
+    f = open(output_file, 'w')
+    f.write(radd_output.decode("utf-8"))
+    f.close()
```
Comment (scopatz) [inline pyne/ensdf_processing.py]: This should be a context manager:

``` python
with open(output_file, 'w') as f:
    f.write(radd_output.decode("utf-8"))
```

File: pyne/ensdf_processing.py [hunk 5]
```diff
@@ -104,3 +104,105 @@ def gabs(inputdict_unchecked):
...
+    inputdict = {}
+    atomic_number = inputdict_unchecked['atomic_number']
+    neutron_number = inputdict_unchecked['neutron_number']
+    output_file = inputdict_unchecked['output_file']
+
+    # Create symlinks to the two binaries the radd executables uses.
+    AK04_path = path_to_exe('98AK04.in')
+    ELE_path = path_to_exe('ELE.in')
+    AK04_set = False
+    ELE_set = False
```
Comment (scopatz) [inline pyne/ensdf_processing.py]: PEP8 don't mix case for variable names.

**Comment by scopatz** [comment]
Hi @howland! Thanks a million for putting this in! It generally looks great, just some stylistic issues to clean up.

File: pyne/ensdf_processing.py (hunk 2 above)
Comment (howland) [inline pyne/ensdf_processing.py:145]: @scopatz The executable ask for a bunch of info via command line, but they can't be passed in as arguments.  I thought I had to use Popen to be able to communicate with the executable?  I also saw the following on:

```
Note Do not use stderr=PIPE with this function as that can deadlock based on the child process error volume. Use Popen with the communicate() method when you need a stderr pipe.
```

https://docs.python.org/2/library/subprocess.html
Do you know of a good way for me to communicate with the executable using subprocess.check_output()?  Sorry if this response is particularly naive, I don't have much experience working with subprocess.

**Comment by howland** [comment]
@scopatz no problem, I addressed your comments here and in #809 on both this and #809.

File: pyne/ensdf_processing.py (hunk 2 above)
Comment (scopatz) [inline pyne/ensdf_processing.py:145]: Oh nm.  This is correct in this case.  Sorry to cause extra noise.

File: pyne/ensdf_processing.py [hunk 6]
```diff
@@ -104,3 +112,117 @@ def gabs(inputdict_unchecked):
...
+    Everything in input dictionary is returned if RULER completes successfully.
+    """
+    inputdict = {}
+    input_file = inputdict_unchecked['input_file']
+    output_report_file = inputdict_unchecked['output_report_file']
+    mode_of_operation = inputdict_unchecked['mode_of_operation']
+    assumed_dcc_theory = inputdict_unchecked['assumed_dcc_theory']
+    
+    exe_path = path_to_exe('ruler')
+    ruler_output = subprocess.Popen([exe_path],stdout=subprocess.PIPE,stdin=subprocess.PIPE)
```
Comment (scopatz) [inline pyne/ensdf_processing.py:228]: PEP8 single space after commas

**Comment by scopatz** [comment]
Thanks @howland, some more nitpicky doc issues, but it otherwise looks good to go.

**Comment by howland** [comment]
@scopatz Revised, but I'll go through again and check the style guide tonight.

**Comment by scopatz** [comment]
OK! Ping whenever you want me to check it out.

**Comment by scopatz** [comment]
This looks good to me now, btw.

**Comment by howland** [comment]
@scopatz I think its ready!

**Comment by scopatz** [comment]
Looks good to me boss!

**Pull request merged**